CMakeLists.txt

```diff
@@ -102,15 +110,62 @@ if(GC_ENABLE_BINDINGS_PYTHON)
   message(STATUS "Enabling Python API")
   add_subdirectory(python)
 endif()
+add_subdirectory(test)
 
-set(GC_LIB_LINKED_LIBS
-        GCJitWrapper
-        GCCpuRuntime
+# Export the targets
+add_library(GcInterface INTERFACE)
+target_include_directories(GcInterface INTERFACE
+        $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
+        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:include>
+)
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/ TYPE INCLUDE)
+install(DIRECTORY ${PROJECT_BINARY_DIR}/include/ TYPE INCLUDE
+        PATTERN "CMake*" EXCLUDE PATTERN "*.cmake" EXCLUDE)
+install(TARGETS
+        GcCpuRuntime
+        GcGpuPasses
+        GcInterface
+        GcJitWrapper
+        GcPasses
+        GcUtilsIR
+        MLIRCPURuntimeDialect
+        MLIRCPURuntimeTransforms
+        MLIRLinalgx
+        MLIRMicrokernel
```

Comment:
So these are to cover the static scenario. I think this should eventually be passed through via a global property to be flexible.

Reply:
Do you mean so that each module adds its targets to the global property?

Reply:
Yup, this is just a thought.